ambry-router/src/test/java/com.github.ambry.router/GetManagerTest.java

```diff
@@ -245,23 +246,35 @@ public void testFailureOnAllPollThatSends()
     setOperationParams(chunkSize);
     String blobId = router.putBlob(putBlobProperties, putUserMetadata, putChannel).get();
     mockSelectorState.set(MockSelectorState.ThrowExceptionOnSend);
+    FutureResult futureResult;
     try {
-      router.getBlobInfo(blobId).get();
+      futureResult = (FutureResult) router.getBlobInfo(blobId);
+      while (!futureResult.isDone()) {
+        mockTime.sleep(routerConfig.routerRequestTimeoutMs + 1);
+        Thread.yield();
```

Comment:
as discussed, this is just to yield to let another thread run and possibly make progress since the intention is not to continuously increment the time.